### PR TITLE
Adds support for Parameter based min & max flow in AggregatedNode

### LINKED ISSUE #461: AggregatedNode min & max flow can not be parameters
#174 added support for min and max flow constraints on `AggregatedNode`. However these are constants and can not be set with a `Parameter`.

### PR DESCRIPTION
**Pull request by jetuk**
- Implemented in GLPK solver only
- Small refactor of the aggregated nodes in the solver
- Found a small bug in the setting of `None` to min_flow & max_flow on Node

Fixes #461

### CODE REVIEW

File: pywr/_core.pyx [hunk 1]
```diff
@@ -591,37 +597,76 @@ cdef class AggregatedNode(AbstractNode):
...
+            if self._min_flow_param is None:
+                return self._min_flow
+            return self._min_flow_param
+
         def __set__(self, value):
             if value is None:
-                value = inf
-            self._max_flow = value
-            self.model.dirty = True
+                self._min_flow = -inf
```
Comment (snorfalorpagus) [inline pywr/_core.pyx]: Why is this `= -inf` here but `= 0` previously (L466)

File: pywr/solvers/cython_glpk.pyx [hunk 2]
```diff
@@ -321,7 +313,7 @@ cdef class CythonGLPKSolver:
                 ind[1+i] = 1+col
                 val[1+i] = 1.0
             glp_set_mat_row(self.prob, row, length, ind, val)
-            glp_set_row_bnds(self.prob, row, constraint_type(min_flow, max_flow), min_flow, max_flow)
+            glp_set_row_bnds(self.prob, row, GLP_FX, 0.0, 0.0)
```
Comment (snorfalorpagus) [inline pywr/solvers/cython_glpk.pyx]: This line isn't needed?

File: tests/test_agg_constraints.py [hunk 3]
```diff
@@ -104,6 +106,27 @@ def test_aggregated_node_max_flow(model):
...
+    """Nodes constrained by the max_flow of their AggregatedNode using a Parameter """
+    A = Input(model, "A", max_flow=20.0, cost=1)
+    B = Input(model, "B", max_flow=20.0, cost=2)
+    Z = Output(model, "Z", max_flow=100, cost=-10)
+
+    A.connect(Z)
+    B.connect(Z)
+
+    agg = AggregatedNode(model, "agg", [A, B])
+    agg.max_flow = ConstantParameter(model, 30.0)
```
Comment (snorfalorpagus) [inline tests/test_agg_constraints.py]: We should test with a parameter that actually changes it's value. I wonder if we should roll these tests up with parameterization?

File: pywr/_core.pyx (hunk 1 above)
Comment (jetuk) [inline pywr/_core.pyx]: It defaulted to -inf in `AggregatedNode` before. Default in `Node` was/is zero.

File: pywr/solvers/cython_glpk.pyx (hunk 2 above)
Comment (jetuk) [inline pywr/solvers/cython_glpk.pyx]: I just followed the pattern from the previous section which includes this line.

File: tests/test_agg_constraints.py (hunk 3 above)
Comment (jetuk) [inline tests/test_agg_constraints.py]: Yes, we should do both. I didn't have time to do anything better.

**Pull request merged**